server/deskroom/knowledge_base/endpoints.py

```diff
@@ -19,8 +29,62 @@ async def get_knowledge_base(
 ) -> list[KnowledgeBase]:
     response = (
         await supabase.table("knowledge_base")
-        .select("*")
+        .select("*, organizations(company_info_policy)")
         .eq("org_key", item.organization_key)
         .execute()
     )
+
     return response.data
+
+
+@router.post("/upload/{org_key}")
+async def make_knowledge_base(
+    org_key: str,
+    file: UploadFile = File(...),
+    supabase: AsyncClient = Depends(create_supabase_async_client),
+) -> list[KnowledgeBase]:
+    questions = []
+    answers = []
+
+    raw_df = pd.read_excel(
+        file.file.read(), engine="openpyxl", sheet_name="Message data"
+    )
+
+    supabase_response = await (
+        supabase.table("organizations")
+        .select("id", "name_eng")
+        .eq("key", org_key)
+        .execute()
+    )
+    if not supabase_response.data:
+        raise ValueError("No ID / Name for given Org Key")
+
+    raw_df = raw_df.dropna()
+    processed_df = await process_raw_file(raw_df)
+
+    discovery_str = await generate_discovery_string(processed_df)
+    company_policy = await create_policy(discovery_str)
+    chat_ids = list(processed_df["chatId"].unique())
+    for chat_id in chat_ids:
+        try:
+            qa_string = await generate_qa_string(processed_df, chat_id)
+            discovered = await create_qa(company_policy, qa_string)
+            discovered_ = literal_eval(discovered)
+            for qa in list(discovered_.values()):
+                questions.append(qa["Question"])
+                answers.append(qa["Answer"])
+        except (ValueError, KeyError):
+            pass
+    update_df = pd.DataFrame({"Question": questions, "Answer": answers})
+    updated_data = []
+    for row_num in range(len(update_df)):
+        updated_files = {
+            "org_id": supabase_response.data[0]["id"],
+            "org_name": supabase_response.data[0]["name_eng"],
+            "org_key": org_key,
+            "question": update_df["Question"].tolist()[row_num],
+            "answer": update_df["Answer"].tolist()[row_num],
+        }
+        data = await supabase.table("knowledge_base").insert(updated_files).execute()
```

Comment:
Insert문은 list로 들어가서 for 문에서 매번 넣는거보다, 한 번에 넣는게 더 좋을거에요 (단, 1000개 데이터 이상 벌크는 오히려 떨어질듯)

```suggestion
    response = await supabase.table("knowledge_base").insert([updated_files for row_num in updated_df]).execute()
```